app/Http/Controllers/Api/InventoryController1.php

```diff
@@ -0,0 +1,51 @@
+<?php namespace App\Http\Controllers;
+
+use App\Http\Requests;
+use App\Http\Controllers\Controller;
+use App\Item;
+
+use Illuminate\Support\Facades\Response;
+use Illuminate\Support\Facades\Input;
+
+class InventoryController1 extends Controller {
+
+	/**
+	 * Display a listing of the resource.
+	 *
+	 * @param  string $query
+	 * @return json all or queried $items
+	 */
+	public function index($query)
+	{
```

Comment:
In this section, we pull in all the items on line 20, then it looks like line 23 is trying to filter them. What we should do is something like:

```
if ( !is_null($query) )
{
    $items = Items::where(...)->get();
}
else
{
    $items = Items::all();
}
```

It's arguable whether we even want to allow them to retrieve an unfiltered list of *all* inventory items. Maybe throw a 400 (bad request) if they don't specify a query?